single_kernel_mongo/managers/sharding.py

```diff
@@ -598,6 +609,16 @@ def synchronise_cluster_secrets(self, relation: Relation, leaving: bool = False)
         # By setting the status we ensure that the former statuses of this component are removed.
         self.state.statuses.set(ShardStatuses.ACTIVE_IDLE.value, scope="unit", component=self.name)
 
+        # Add the certificate if it is present
+        if backup_tls_chain := self.state.shard_state.backup_ca_secret:
+            self.dependent.save_ca_cert_to_trust_store(TrustStoreFiles.PBM, backup_tls_chain)
+        else:
+            # If it is not in the databag, always remove it, it won't change a
+            # thing if the file is not present, remove_ca_cert_from_trust_store will early return.
+            self.dependent.remove_ca_cert_from_trust_store(TrustStoreFiles.PBM)
+        # We updated the configuration, so we restart PBM.
+        self.dependent.backup_manager.configure_and_restart(force=True)
```

Comment:
I don't think we should _always_ restart even when no ca_chain is passed / removed. I would rather we only do it when there is an addition / removal of the CA chain